src/mpid/ch4/netmod/ucx/ucx_recv.h

```diff
@@ -184,37 +181,27 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_mpi_imrecv(void *buf,
                                                             (char *) buf + dt_true_lb,
                                                             data_sz,
                                                             ucp_dt_make_contig(1),
-                                                            MPIDI_UCX_REQ(message).
-                                                            a.message_handler,
+                                                            MPIDI_UCX_REQ(message).message_handler,
                                                             &MPIDI_UCX_mrecv_cmpl_cb);
     } else {
         MPIR_Datatype_ptr_add_ref(dt_ptr);
         ucp_request =
             (MPIDI_UCX_ucp_request_t *) ucp_tag_msg_recv_nb(MPIDI_UCX_global.worker,
                                                             buf, count,
                                                             dt_ptr->dev.netmod.ucx.ucp_datatype,
-                                                            MPIDI_UCX_REQ(message).
-                                                            a.message_handler,
+                                                            MPIDI_UCX_REQ(message).message_handler,
                                                             &MPIDI_UCX_mrecv_cmpl_cb);
     }
     MPIDI_UCX_CHK_REQUEST(ucp_request);
 
-    if (ucp_request->req) {
-        if (unlikely((uint64_t) ucp_request->req == (uint64_t) UCS_ERR_MESSAGE_TRUNCATED)) {
-            message->status.MPI_ERROR = MPI_ERR_TRUNCATE;
-        } else {
-            ucp_tag_recv_info_t *info = ucp_request->req;
-            message->status.MPI_ERROR = MPI_SUCCESS;
-            message->status.MPI_SOURCE = MPIDI_UCX_get_source(info->sender_tag);
-            message->status.MPI_TAG = MPIDI_UCX_get_tag(info->sender_tag);
-            MPIR_STATUS_SET_COUNT(message->status, info->length);
-            MPL_free(ucp_request->req);
-        }
+    if (ucp_request->status) {
+        MPIR_Memcpy(&message->status, ucp_request->status, sizeof(MPI_Status));
```

Comment:
I believe we could do `message->status = * (ucp_request->status);`

Reply:
True. I think I like that better, so I'll make the change.